src/test/java/FruitShopMockSpy.java

```diff
@@ -0,0 +1,117 @@
+import fruitShop.Fruit;
+import fruitShop.FruitShop;
+import fruitShop.FruitType;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static fruitShop.FruitType.*;
+import static org.mockito.Mockito.*;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+
+@RunWith(MockitoJUnitRunner.class)
+public class FruitShopMockSpy {
+
+    @Spy
+    @InjectMocks
+    private FruitShop fruitShop;
+    @Mock
+    private List<Fruit> fruits;
+    @Mock
+    private Fruit apple;
```

Comment:
Why you are creating mock on all Fruit objects? You need to mock just one object Fruit fruit, and create a different realizations of it methods.

Reply:
It is normal. More bigger level (List in our case) also can be mock